pkg/queue/sharedmain/main.go

```diff
@@ -243,52 +243,53 @@ func Main(opts ...Option) error {
 		concurrencyendpoint = queue.NewConcurrencyEndpoint(env.ConcurrencyStateEndpoint, env.ConcurrencyStateTokenPath)
 	}
 
+	type service struct {
+		name string
+		srv  *http.Server
+		tls  bool
+	}
+
+	var srvs []service
```

Comment:
If order is not a problem you could start with:

```
srvs := []service{service{name: "metrics", srv: buildMetricsServer(protoStatReporter), tls: false}}
```

Reply:
since we moved to build* functions creating the service, we can consider this resolved